### PR TITLE
0.5.0 migration guide

### PR DESCRIPTION
**Pull request by drknzz**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

### CODE REVIEW

File: docs/migration_guide.rst [hunk 1]
```diff
@@ -1,3 +1,84 @@
...
+
+Currently, compatible Python versions are ``python3.8`` and ``python3.9``.
+
+New Transaction version
+-----------------------
+
+Cairo 0.10.0 brings new version of the transaction.
+The differences:
+
+- ``contract_address`` field is now called ``account_contract_address``
```
Comment (war-in) [inline docs/migration_guide.rst:65]: It hasn't chanded. It is mentioned in the notion, but there is no difference in the package

File: docs/migration_guide.rst [hunk 2]
```diff
@@ -1,3 +1,84 @@
+0.4.8 Migration guide
+=====================
+
+0.4.8 of starknet.py brings fresh changes including a breaking change in contracts' code.
```
Comment (war-in) [inline docs/migration_guide.rst:4]: Maybe we should mention that this breaking change doesn't come from starknet.py?

File: docs/migration_guide.rst [hunk 3]
```diff
@@ -1,19 +1,68 @@
...
 
-New transaction version
+- new cairo syntax,
+- new transaction version (1),
+- new ``__validate__`` endpoint in accounts.
+
+``starknet.py`` 0.5.0 has an experimental support for new features and tries to minimize number of breaking changes for
+users who want to use the old transaction version (0). Please note that support for this transaction version will be
+removed in the future.
+
```
Comment (Solpatium) [inline docs/migration_guide.rst:13]: Add a note that informs users that they can use the previous version if they don't need any new features.

**Review by Solpatium — CHANGES_REQUESTED** [review]

File: docs/migration_guide.rst [hunk 4]
```diff
@@ -1,19 +1,76 @@
...
+.. note::
+
+    There is no need to upgrade ``starknet.py`` to the newest version because the old one is still compatible with StarkNet.
```
Comment (cptartur) [inline docs/migration_guide.rst:16]: Are we sure about this line?

**Review by Solpatium — APPROVED** [review]

**Review by cptartur — COMMENTED** [review]
One question, otherwise looks very good

**Pull request merged**